src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -104,7 +106,15 @@ public void save(
 
         if (!FeaturesUtils.isOnlineDQ(updatedClaim) && isRejectResponseNoMediation(claimantResponse)) {
             directionsQuestionnaireService.updateDirectionsQuestionnaireDeadline(
-                updatedClaim, LocalDateTime.now(), authorization);
+                    updatedClaim, LocalDateTime.now(), authorization);
+            updatedClaim = claimService.getClaimByExternalId(externalId, authorization);
+        }
+
+        if (FeaturesUtils.isOnlineDQ(updatedClaim)) {
+            appInsights.trackEvent(AppInsightsEvent.BOTH_PARTIES_ONLINE_DQ, REFERENCE_NUMBER, updatedClaim.getReferenceNumber());
+            updatedClaim = claimService.getClaimByExternalId(externalId, authorization);
+        } else {
+            appInsights.trackEvent(AppInsightsEvent.BOTH_PARTIES_OFFLINE_DQ, REFERENCE_NUMBER, updatedClaim.getReferenceNumber());
```

Comment:
Raising the app insights events should been done in the raiseAppInsightsEvents method. Also the calls to `getClaimByExternalId` aren't needed

Reply:
I'll remove it